lib/jackal-kitchen/tester.rb

```diff
@@ -129,57 +133,78 @@ def execute(msg)
         end
       end
 
-      # Create a remote instance and return information for accessing it via SSH
-      # TODO: Actually provision instances. For now, read ssh connection params from config.
-      #
-      # @param instance_config [Hash]
-      # @returns [Smash]
-      def provision_instance(instance_config = {})
-        config.fetch(:ssh, {})
+      # Return a Rye::Box object for the described instance
+      # @param state [Hash]
+      # @return [Rye::Box]
+      def instance_ssh_connection(state)
+        Rye::Box.new(
+          state[:hostname],
+          :port => state.fetch(:port, 22) ,
+          :user => state[:username],
+          :keys => state.fetch(:ssh_key, nil),
+          :password => 'invalid',
+          :password_prompt => false
+        )
+      end
+
+      # Upload the teapot-handler cookbook fixture to a remote instance
+      # @param remote [Rye::Box]
+      # @return [TrueClass]
+      def insert_teapot_cookbook(remote)
+        # inserting teapot
+        # upload file
+        # mkdir -p /tmp/kitchen/cookbooks
+        # unpack cookbook into /tmp/kitchen/cookbooks
+        # get cookbook into run list???
+        info("Inserting teapot-handler cookbook into kitchen cookbook path on #{remote.host}")
+        remote.file_upload(File.join(File.dirname(__FILE__), %w(.. .. data teapot-handler-cookbook.tar)), '/tmp')
+        remote.disable_safe_mode
+        remote.exec('mkdir -p /tmp/kitchen/cookbooks/teapot-handler')
+        remote.exec('chmod -R a+rwx /tmp/kitchen')
+        remote.cd('/tmp/kitchen/cookbooks/teapot-handler')
+        remote.exec('tar --strip-components=1 -xvf /tmp/teapot-handler-cookbook.tar')
       end
 
       # Write .kitchen.local.yml overrides into specified path
       #
       # @param path [String]
-      # @param instance [Hash]
-      def insert_kitchen_local(path, instance = {})
+      # @param instance [Miasma::Compute::Server]
+      # @return [TrueClass]
+      def insert_kitchen_local(path)
+
+        platforms = kitchen_platforms(path)
+
         File.open(File.join(path, '.kitchen.local.yml'), 'w') do |file|
           file.puts '---'
```

Comment:
super nitpicky, but If we to_yaml this content and write it out rather than adding strings to the file, we might be a bit safer & catch potential errors in the yaml writing stage rather than later on then the kitchen run does something weird, in the case where one of the values we're interpolating potentially creates invalid / bad syntax in the yml file